### PR TITLE
Add workaround for ambiguous complex multiplication overloads in HIP in `larft.cu`

### PR DESCRIPTION
**Pull request by msimberg**

### CODE REVIEW

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
This is not ready for merging since it depends on #1240 which needs some cleanup. It's ready for review though (the last commit is the only new change in this PR). This implements the most minimal workaround for the issue. On #1240 the ROCm 6.0.2 pipeline fails with the error reported in #1239 (https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/8655570191#L231). On this PR the pipeline passes. The ROCm 6.2.4 pipeline fails on both this and #1240, and I would suggest skipping it at least for the 0.7.1 patch release (https://github.com/eth-cscs/DLA-Future/pull/1240#issuecomment-2547815399).

**Comment by msimberg** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Review by albestro — APPROVED** [review]

File: src/lapack/gpu/larft.cu [hunk 1]
```diff
@@ -8,6 +8,9 @@
 // SPDX-License-Identifier: BSD-3-Clause
 //
 
+#include <complex>
```
Comment (RMeli) [inline src/lapack/gpu/larft.cu:11]: Looks like we are missing `std::complex` from `inspect.toml`.

**Review by RMeli — APPROVED** [review]

File: src/lapack/gpu/larft.cu (hunk 1 above)
Comment (msimberg) [reply src/lapack/gpu/larft.cu:11]: Excellent point. I'm fixing those in #1243. There's a long list of includes missing...

**Comment by msimberg** [comment]
With #1240 merged, this PR fixes the build error on the new CI configuration: https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/8672697339.

**Pull request merged**